rfcs/0011-separate-flyway-from-pipeline.md

```diff
@@ -0,0 +1,382 @@
+</details>
+
+* A new migrator service account and associated migrator database role that has all required privileges for running
+migrations and hibernate validation will be created.
+
+* The Jenkins APP AMI build phase will build a new migrator AMI that includes the new application.
+
+* The migrator AMI and migrator service account will be used to run the migrator application on a new EC2 instance as
+part of a new Jenkins pipeline stage that will be inserted prior to each of the three stages that deploy to TEST,
+PROD-SBX, PROD.
+
+* The Jenkins pipeline will not proceed to the next stage (deploying the other applications) until the migrator
+application has exited with a status indicating success. In the event that this application status indicates that not
+all operations were successful, the Jenkins deployment will halt and be marked as a failure.
+
+* The new application will produce log files that contain the Flyway and Hibernate logging that is currently captured in
+the BFD Pipeline application logs. These new application logs will be sent to Splunk and Cloudwatch.
+
+* The migrator EC2 instance could be powered off or terminated once the migrator application exits and all artifacts
+have been gathered.
+
+New Flyway migration once the above is in place:
+
+* A new Flyway migration will be developed that creates all non-user roles and sets privileges correctly on all existing
+database objects.
+
+### Proposed Solution: Unresolved Questions
+[Proposed Solution: Unresolved Questions]: #proposed-solution-unresolved-questions
+
+Do we need to continue to invoke Hibernate validation at all? What benefit do we derive from this check that is not
+already derived from running the unit and integration tests? What sorts of problems would not be caught by the tests
+that would be caught by hibernate validation?
+
+### Proposed Solution: Drawbacks
+[Proposed Solution: Drawbacks]: #proposed-solution-drawbacks
+
+This adds time to the deployment, currently estimated at 10-12 minutes total.
+
+This makes the Jenkins deployment more complicated and adds another failure mode.
+
+This makes the application more complicated and introduces a new point of failure.
+
+This makes debugging more complex. There will be new places (log files, possibly directories) where debugging
+information will be kept.
+
+### Proposed Solution: Notable Alternatives
+[Proposed Solution: Notable Alternatives]: #proposed-solution-notable-alternatives
+
+Instead of running Flyway and Hibernate validation in a new application, Flyway could be invoked from the command line
+instead. These leaves the question as to where to invoke Hibernate validation. If it continues to be invoked in the
+applications, it is possible to encounter validation errors if auto-scaling of an old application occurs just after a
+new schema has been deployed. One possible answer is to not invoke Hibernate validation at all (see open questions).
```

Comment:
I like the idea of using a custom migration app to trigger the Flyway migration and do the hibernate validation.  It should be a very simple app and it ensures everything is done with the same configuration that the pipeline and api apps will be using.